shared/lib_battery_lifetime_nmc.cpp

```diff
@@ -82,33 +86,23 @@ lifetime_t * lifetime_nmc_t::clone() {
     return new lifetime_nmc_t(*this);
 }
 
-//calculate_Uneg, and calculate_Voc are picewise linear functions. The linear functions were obtained by using values
-//from  Table B.1 of the applied energy paper " Analysis of degradation in residential battery energy storate systems
-// for rate-based use-cases"
-
 double lifetime_nmc_t::calculate_Uneg(double SOC) {
-    double Uneg;
-    if (SOC <= 0.1)
-        Uneg = ((0.2420 - 1.2868) / 0.1) * SOC + 1.2868;
-    else
-        Uneg = ((0.0859 - 0.2420) / 0.9) * (SOC - 0.1) + 0.2420;
+    size_t prev_ind = floor(SOC * 10);
+    size_t next_ind = prev_ind + 1;
+    double Uneg = unegs[prev_ind] + (unegs[next_ind] - unegs[prev_ind]) / 0.1 * (SOC - (double)prev_ind / 10);
```

Comment:
This crashes at SOC of 100% - prev_ind ends up being 10, which pushes next_ind to 11, which is out of bounds. It looks like calculate_Voc will have the same problem.

Reply:
Good catch, thanks

Reply:
When I run the tests locally, the fix has the same problem. It looks like line 93 should be `>=`

Reply:
Yes, thanks! Updated.